File: backend/src/main/java/com/workforcex/backend/dto/WorkerProfileRequest.java
package com.workforcex.backend.dto;

import java.time.LocalDate;

/**
 * What a worker can submit to create/update their profile.
 * All fields optional, per registration philosophy.
 */
public record WorkerProfileRequest(
        String name,
        String gender,
        LocalDate dateOfBirth,
        String email,
        String address,
        String city,
        String state,
        String skills,
        Integer experience,
        Double preferredSalary
) {
}

File: android/worker/shared/src/commonMain/kotlin/com/workforcex/shared/models/WorkerProfileRequest.kt
package com.workforcex.shared.models

data class WorkerProfileRequest(
    val name: String,
    val gender: String,
    val city: String,
    val state: String,
    val skills: String,
    val experience: Int,
    val preferredSalary: Double,
    val availability: String,
    val languages: String,
    val description: String
)